added log4js for better debugging and /shelters/{shelterID}/pets

dotte-osu opened the pull request.

dotte-osu committed:
added db.config.js to .gitignore

dotte-osu committed:
added log4js

dotte-osu committed:
added newline

dotte-osu committed:
added logger

dotte-osu committed:
added log.error

dotte-osu committed:
added /shelters/{shelterID}/pets

dotte-osu commented:
- added log4js
- added /shelters/{shelterID}/pets

asching7108 commented on src/controllers/sheltersController.js:
```diff
@@ -48,23 +52,25 @@ const postShelters = async (req, res, next) => {
 };
 
 const updateShelter = async (req, res, next) => {
+  log.debug("Calling getShelters...Verifying user inputs...");
```
It should be updateShelters here

dotte-osu replied:
thank you for catching it!

asching7108 commented on src/server.js:
```diff
@@ -27,6 +30,7 @@ router.post('/shelters', sheltersController.postShelters);
 router.get('/shelters/:shelterID', sheltersController.getShelter);
 router.delete('/shelters/:shelterID', sheltersController.deleteShelter);
 router.patch('/shelters/:shelterID', sheltersController.updateShelter);
+router.get('/shelters/:shelterID/pets', sheltersController.getPets);
```
Just a side note, the getShelters route is still /shelters/test, do we want to remove the /test?

dotte-osu replied:
Removed the test since we are not using it.

asching7108 commented on src/controllers/sheltersController.js:
```diff
@@ -87,18 +93,33 @@ const updateShelter = async (req, res, next) => {
+  log.debug("Calling getPets...");
+  await getAllPets(req.params.shelterID)
+    .then((dbResponse) => {
+      if (dbResponse.length == 0) res.sendStatus(404);
```
I wonder if we would want to distinguish 'shelter not found' or 'shelter exists but has no pets yet' here. It depends on how we want to implement the frontend, but I imagine we may need a way to tell the difference? Like if the shelter id doesn't exist, we want to return an error or redirect the user to the not found page, but if the shelter simply doesn't have any pets yet we may want to just display an empty list and a create button. If we want to tell the difference we'll probably need to run a select query to the shelters table first in the model. Let me know what you think!

dotte-osu replied:
I agree, I changed the code to return below.
 'shelter not found': --> 404, "Shelter not found"
 'shelter exists but has no pets yet' --> 200. [] (empty array)

asching7108 reviewed (commented):
Thank you for adding log4js!! It definitely will help a lot. I don't have a lot to add but mainly just a question about the getPets endpoint. All looks really good!

dotte-osu committed:
fixed debug message, fixed getPets(), added tests

dotte-osu committed:
removed temp code

dotte-osu commented:
Thank you for catching the small things. I fixed the debug comment, removed the sample route. I agree that we should distinguish 'shelter not found' or 'shelter exists but has no pets yet'. I changed the responses for it.

dotte-osu committed:
fixed comments

dotte-osu commented:
Thank you for catching the small things. I fixed the debug comment, removed the sample route. I agree that we should distinguish 'shelter not found' or 'shelter exists but has no pets yet'. I changed the responses for it.
